tests/integration/ha_tests/helpers.py

```diff
@@ -463,3 +470,94 @@ async def update_restart_condition(ops_test: OpsTest, unit, condition: str):
+    for line in proc.splitlines():
+        if "Storage" in line:
+            continue
+
+        if len(line) == 0:
+            continue
+
+        if "detached" in line:
+            continue
+
+        unit_name = line.split()[0]
+        app_name = unit_name.split("/")[0]
+        if app_name == app:
+            return line.split()[3]
+
+
+def storage_id(ops_test, unit_name):
+    """Retrieves  storage id associated with provided unit.
+
+    Note: this function exists as a temporary solution until this issue is ported to libjuju 2:
+    https://github.com/juju/python-libjuju/issues/694
+    """
+    model_name = ops_test.model.info.name
+    proc = subprocess.check_output(f"juju storage --model={model_name}".split())
+    proc = proc.decode("utf-8")
+    for line in proc.splitlines():
+        if "Storage" in line:
+            continue
+
+        if len(line) == 0:
+            continue
+
+        if "detached" in line:
+            continue
+
+        if line.split()[0] == unit_name:
+            return line.split()[1]
+
+
+async def add_unit_with_storage(ops_test, app, storage):
+    """Adds unit with storage.
+
+    Note: this function exists as a temporary solution until this issue is resolved:
+    https://github.com/juju/python-libjuju/issues/695
+    """
+    expected_units = len(ops_test.model.applications[app].units) + 1
+    prev_units = [unit.name for unit in ops_test.model.applications[app].units]
+    model_name = ops_test.model.info.name
+    add_unit_cmd = f"add-unit {app} --model={model_name} --attach-storage={storage}".split()
+    return_code, _, _ = await ops_test.juju(*add_unit_cmd)
+    assert return_code == 0, "Failed to add unit with storage"
+    await ops_test.model.wait_for_idle(apps=[app], status="active", timeout=1000)
+    assert (
+        len(ops_test.model.applications[app].units) == expected_units
+    ), "New unit not added to model"
+
+    # verify storage attached
+    curr_units = [unit.name for unit in ops_test.model.applications[app].units]
+    new_unit = list(set(curr_units) - set(prev_units))[0]
+    assert storage_id(ops_test, new_unit) == storage, "unit added with incorrect storage"
+
+    # return a reference to newly added unit
+    for unit in ops_test.model.applications[app].units:
+        if unit.name == new_unit:
+            return unit
+
+
+async def reused_storage(ops_test: OpsTest, unit_name, removal_time) -> bool:
+    """Returns True if storage provided to Postgresql has been reused.
+
+    Checks Patroni logs for when the database was stopped.
+    """
+    logs = await run_command_on_unit(
+        ops_test, unit_name, "journalctl -u snap.charmed-postgresql.patroni.service"
+    )
+    if match := reuse_vol_regex.search(logs):
+        logs_date = datetime.fromisoformat("T".join(match.groups()))
+        # Log time is within 15 minuntes of the recorded removal time
+        return logs_date + timedelta(minutes=15) >= removal_time
+    return False
```

Comment:
Check for db system interrupted within 15 mins of the unit destroy timestamp. Is there a better way to check?